bossutils/vault.py

```diff
@@ -22,11 +22,28 @@
 import hvac
 from . import configuration
 from . import utils
+import functools
+from bossutils.logger import BossLogger
 
 VAULT_SECTION = "vault"
 VAULT_URL_KEY = "url"
 VAULT_TOKEN_KEY = "token"
 
+def catch_expire(function):
+    """Catches error raised when token has expired and rotates token if so.
+    Executes the same function once the token has been replaced."""
+    def wrapper(*args, **kwargs):
+        try:
+            return function(*args, **kwargs)
+        except Exception as e:
+            blog = BossLogger().logger
+            blog.info(str(e))
+            msg = "Your token had expired.  Dynamically creating a new one."
+            blog.info(msg)
+            Vault.login(self=None)
```

Comment:
Not sure how this even passed testing. Passing `None` for self should have caused an exception. `Vault.login` uses `self.config` and `self.client`.

You need to pull the `self` value out of `args` or just make the definition `def wrapper(self, *args, **kwargs)` and make sure to pass `self` in the call to `function(self, *args, **kwargs)`.